src/main/java/com/google/devtools/build/lib/analysis/Runfiles.java

```diff
@@ -980,6 +980,13 @@ private Builder addTargetIncludingFileTargets(TransitiveInfoCollection target,
         return this;
       }
 
+      if (mapping == RunfilesProvider.DATA_RUNFILES) {
+        FileProvider fileProvider = target.getProvider(FileProvider.class);
```

Comment:
Can you remove the branch above L#972 to L#981, but keep the TODO comment? I believe the new branch completely covers it.

The case there's no RunfilesProvider should not raise any exceptions. Also calling `addTargetExceptFileTargets` where there's no RunfilesProvider shouldn't make a difference.